torchmetrics/collections.py

```diff
@@ -149,20 +157,25 @@ def update(self, *args, **kwargs):  # pylint: disable=E0202
             m.update(*args, **m_kwargs)
 
     def compute(self) -> Dict[str, Any]:
-        return {self._set_prefix(k): m.compute() for k, m in self.items()}
+        return {self._set_name(k): m.compute() for k, m in self.items()}
 
     def reset(self) -> None:
         """ Iteratively call reset for each metric """
         for _, m in self.items():
             m.reset()
 
-    def clone(self, prefix: Optional[str] = None) -> 'MetricCollection':
+    def clone(self, prefix: Optional[str] = None, postfix: Optional[str] = None) -> 'MetricCollection':
         """ Make a copy of the metric collection
         Args:
             prefix: a string to append in front of the metric keys
+            postfix: a string to append after the keys of the output dic
+
         """
         mc = deepcopy(self)
-        mc.prefix = self._check_prefix_arg(prefix)
+        if prefix is not None:
+            mc.prefix = self._check_arg(prefix, 'prefix')
+        if postfix is not None:
+            mc.postfix = self._check_arg(postfix, 'postfix')
```

Comment:
```suggestion
        if prefix:
            mc.prefix = self._check_arg(prefix, 'prefix')
        if postfix:
            mc.postfix = self._check_arg(postfix, 'postfix')
```
an empty string is also not welcome...